aiodiscover/discovery.py

```diff
@@ -139,11 +139,19 @@ async def _async_get_nameservers(
         """Get nameservers to query."""
         all_nameservers = list(sys_network_data.nameservers)
         router_ip = sys_network_data.router_ip
-        assert router_ip is not None
```

Comment:
`router_ip:` is typed to `str | None` so the assertion needs to stay for the type checker